src/main/groovy/pl/allegro/tech/build/axion/release/domain/VersionResolver.groovy

```diff
@@ -44,27 +52,94 @@ class VersionResolver {
         GitRepository repository = (GitRepository) this.repository
         Pattern releaseTagPattern = ~/^${tagProperties.prefix}.*/
         Pattern nextVersionTagPattern = ~/.*${nextVersionProperties.suffix}$/
-
-        TagsOnCommit tags = repository.latestTags(releaseTagPattern)
-        TagsOnCommit previousTags = tags;
-
-        Map currentVersionInfo = versionFromTags(tags.tags, false, nextVersionTagPattern, versionFactory)
-        Version currentVersion = currentVersionInfo.version
-
-
+        
+        Map currentVersionInfo = null
+        Map previousVersionInfo = null
+        List<TagsOnCommit> allTaggedCommits = repository.allTaggedCommits(releaseTagPattern, null, true, true)
+        currentVersionInfo = versionFromTaggedCommits(allTaggedCommits, false, nextVersionTagPattern, versionFactory)
+        boolean isHead = currentVersionInfo.isHead
+        
+        
+        TagsOnCommit previousTags = repository.latestTags(releaseTagPattern)
         while (previousTags.hasOnlyMatching(nextVersionTagPattern)) {
-            previousTags = repository.latestTags(releaseTagPattern, previousTags.commitId)
+          previousTags = repository.latestTags(releaseTagPattern, previousTags.commitId)
         }
-        Version previousVersion = versionFromTags(previousTags.tags, true, nextVersionTagPattern, versionFactory).version
-
+        previousVersionInfo = versionFromTags(previousTags.tags, true, nextVersionTagPattern, versionFactory)
+        
+        Version currentVersion = currentVersionInfo.version
+        Version previousVersion = previousVersionInfo.version
         return [
-                current         : currentVersion,
-                previous        : previousVersion,
-                onReleaseTag    : tags.isHead && !currentVersionInfo.isNextVersion,
-                onNextVersionTag: currentVersionInfo.isNextVersion,
-                noTagsFound     : currentVersionInfo.noTagsFound
+          current         : currentVersion,
+          previous        : previousVersion,
+          onReleaseTag    : isHead && !currentVersionInfo.isNextVersion,
+          onNextVersionTag: currentVersionInfo.isNextVersion,
+          noTagsFound     : currentVersionInfo.noTagsFound
         ]
     }
+    
+    private Map readVersionsByHighestVersion(VersionFactory versionFactory,
+                                              TagProperties tagProperties,
+                                             NextVersionProperties nextVersionProperties) {
+        GitRepository repository = (GitRepository) this.repository
```

Comment:
okay, i will take a look at it myself, might be something that i missed when changing this code for 1.5.0